dateparser/custom_lang_detect/__init__.py

```diff
@@ -0,0 +1,20 @@
+import importlib
+from dateparser.conf import apply_settings
+
+
+class CustomLanguageDetectCache:
+    loaded_parser = None
+    detect_languages_module = None
+
+@apply_settings
+def detect_languages(settings=None):
+  if not CustomLanguageDetectCache.loaded_parser:
```

Comment:
I don't think it's a sound method to cache this - if someone passes changed settings, we'll get old results. Why do we need our own caching here? importlib already has a cache.

Reply:
Right I will resolve it